### PR TITLE
Add MobileNetV3 architecture for Segmentation

### PR DESCRIPTION
**Pull request by datumbox**
Adding MobileNetV3 models for Semantic Segmentation (resolution 520):

## Lite R-ASPP with Dilated MobileNetV3 Large Backbone

Heavily optimized for speed. Good for actual mobile usage.

Weight checkpoint:
```
PR3276/3rd_training/35354083/model_28.pth
```

Validate:
```
python -m torch.distributed.launch --nproc_per_node=2 --use_env train.py --dataset coco\
   --model lraspp_mobilenet_v3_large --test-only --pretrained
```

Accuracy metrics:
```
0: global correct: 91.2
0: average row correct: ['94.5', '84.3', '69.5', '72.8', '57.7', '42.0', '77.0', '57.0', '90.4', '36.1', '76.0', '60.8', '81.4', '78.9', '81.0', '87.6', '51.3', '83.9', '62.2', '84.2', '56.1']
0: IoU: ['90.2', '69.2', '57.7', '58.5', '47.8', '35.7', '69.5', '47.1', '79.1', '29.6', '62.6', '34.2', '65.5', '63.4', '70.0', '76.8', '30.1', '61.9', '46.8', '70.6', '49.1']
0: mean IoU: 57.9
```

Speed Benchmark: `0.3278 sec per image on CPU`

## DeepLabV3 with Dilated MobileNetV3 Large Backbone

Offers good balance between speed and accuracy, significantly faster than the FCN model with a resnet50 backbone without sacrificing too much accuracy.

Weight checkpoint:
```
PR3276/3rd_training/35354080/model_28.pth
```

Validate:
```
python -m torch.distributed.launch --nproc_per_node=2 --use_env train.py --dataset coco\
   --model deeplabv3_mobilenet_v3_large --test-only --pretrained
```

Accuracy metrics:
```
0: global correct: 91.2
0: average row correct: ['93.7', '84.9', '73.6', '74.6', '63.6', '50.6', '80.7', '65.1', '91.3', '42.2', '80.4', '70.6', '82.4', '81.8', '83.7', '88.5', '52.6', '87.8', '65.9', '88.3', '63.3']
0: IoU: ['90.1', '69.7', '58.2', '61.3', '49.7', '37.6', '72.7', '52.7', '79.1', '32.2', '64.6', '36.2', '66.7', '67.4', '70.1', '77.3', '33.1', '67.8', '51.1', '73.3', '54.4']
0: mean IoU: 60.3
```

Speed Benchmark: `0.5869 sec per image on CPU`

### CODE REVIEW

File: torchvision/models/mobilenetv3.py [hunk 1]
```diff
@@ -82,7 +84,7 @@ def __init__(self, cnf: InvertedResidualConfig, norm_layer: Callable[..., nn.Mod
 
         self.block = nn.Sequential(*layers)
         self.out_channels = cnf.out_channels
-        self.is_strided = cnf.stride > 1
+        self._is_cn = cnf.stride > 1
```
Comment (fmassa) [inline torchvision/models/mobilenetv3.py:87]: out of curiosity, what does `cn` mean in here

File: torchvision/models/mobilenetv3.py [hunk 2]
```diff
@@ -194,78 +196,74 @@ def _mobilenet_v3(
...
 
     Args:
         pretrained (bool): If True, returns a model pre-trained on ImageNet
         progress (bool): If True, displays a progress bar of the download to stderr
-        reduced_tail (bool): If True, reduces the channel counts of all feature layers
-            between C4 and C5 by 2. It is used to reduce the channel redundancy in the
-            backbone for Detection and Segmentation.
     """
+    # non-public config parameters
+    reduce_divider = 2 if kwargs.pop('_reduced_tail', False) else 1
```
Comment (fmassa) [inline torchvision/models/mobilenetv3.py:209]: Is this feature used in any of the models? Otherwise we can just remove it

**Review by fmassa — APPROVED** [review]
Looks great, thanks!

I only have a couple of minor (non-blocking) comments. The only thing I would really like to see fixed before merge is to have the correct `python -m torch.distributed.launch ...` commands for reproducibility.

File: references/segmentation/README.md [hunk 3]
```diff
@@ -31,3 +31,13 @@ python -m torch.distributed.launch --nproc_per_node=8 --use_env train.py --lr 0.
 ```
 python -m torch.distributed.launch --nproc_per_node=8 --use_env train.py --lr 0.02 --dataset coco -b 4 --model deeplabv3_resnet101 --aux-loss
 ```
+
+## deeplabv3_mobilenet_v3_large
+```
+sbatch launch_job.sh --dataset coco -b 4 --model deeplabv3_mobilenet_v3_large --aux-loss --wd 0.000001
```
[the remark this answers is not in the document]
Comment (datumbox) [reply references/segmentation/README.md:37]: Good catch done in both

File: torchvision/models/mobilenetv3.py (hunk 1 above)
Comment (datumbox) [reply torchvision/models/mobilenetv3.py:87]: It's from the C0,C1...C5,Cn names used in Object Detection. I use this feature internally to find out where the downsampling was supposed to happen but it's not always done with strides so I had to rename it. If you have any better name for it, happy to change it. I could not think of any...

File: torchvision/models/mobilenetv3.py (hunk 2 above)
Comment (datumbox) [reply torchvision/models/mobilenetv3.py:209]: This is a unique implementation detail from the paper on MobileNetV3 models and it's supposed to produce a further speed optimization on object detection and segmentation. In our training scripts we don't use it because we do transfer learning from ImageNet but if someone really wants to train it from scratch and go smaller I provide a way to do it.

On current master this is public (see `reduced_tail` param) but here I decide to hide before the release and make it an internal implementation detail for future models. Not quite convinced we will use it but want to provide an implementation very close to the paper.

Personally I would prefer to keep it hidden for now and decide later whether we want this gone. Let me know.

File: torchvision/models/mobilenetv3.py (hunk 1 above)
Comment (fmassa) [reply torchvision/models/mobilenetv3.py:87]: thanks for the explanation. Given that this is private I'm fine with this name

File: torchvision/models/mobilenetv3.py (hunk 2 above)
Comment (fmassa) [reply torchvision/models/mobilenetv3.py:209]: Sounds good, I'm ok keeping this private for now and maybe removing it from the future.
